overlord/snapstate/snapstate_update_test.go

```diff
@@ -4329,6 +4329,105 @@ func (s *snapmgrTestSuite) TestUpdateMany(c *C) {
 	checkIsAutoRefresh(c, ts.Tasks(), false)
 }
 
+func (s *snapmgrTestSuite) TestUpdateManyFailureDoesntUndoSnapdRefresh(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	r := snapstatetest.MockDeviceModel(ModelWithBase("core18"))
+	defer r()
+
+	snapstate.Set(s.state, "some-snap", &snapstate.SnapState{
```

Comment:
for completeness we want some-base to be the base of some-snap in this case, I think  using tracking channel channel-for-base/stable should achieve that